test/gtest/ucp/test_ucp_proto.cc

```diff
@@ -363,23 +370,69 @@ class test_proto_perf : public ucp_test {
 protected:
     virtual void cleanup() override
     {
-        if (m_perf != nullptr) {
-            ucp_proto_perf_destroy(m_perf);
-        }
+        m_perf.reset();
+    }
+
+    static perf_ptr_t create()
+    {
+        ucp_proto_perf_t *perf;
+        ucs_status_t status = ucp_proto_perf_create("test", &perf);
+        ASSERT_UCS_OK(status);
+        return make_perf(perf);
+    }
+
+    static perf_ptr_t from_caps(const ucp_proto_caps_t &caps)
+    {
+        ucp_proto_perf_t *perf;
+        ucs_status_t status = ucp_proto_perf_from_caps("test", &caps, &perf);
+        ASSERT_UCS_OK(status);
+        return make_perf(perf);
+    }
+
+    perf_ptr_t aggregate(const std::vector<perf_ptr_t> &perfs)
+    {
+        std::vector<ucp_proto_perf_t*> perfs_vec;
+        std::transform(perfs.begin(), perfs.end(),
+                       std::back_inserter(perfs_vec),
+                       [](const perf_ptr_t &p) { return p.get(); });
+
+        ucp_proto_perf_t *perf;
+        ucs_status_t status = ucp_proto_perf_aggregate("aggregate",
+                                                       perfs_vec.data(),
+                                                       perfs_vec.size(), &perf);
+        ASSERT_UCS_OK(status);
+        return make_perf(perf);
+    }
+
+    void add_func(size_t start, size_t end,
+                  ucp_proto_perf_factor_id_t factor_id, ucs_linear_func_t func)
+    {
+        add_func(m_perf, start, end, factor_id, func);
+    }
+
+    void add_func(perf_ptr_t perf, size_t start, size_t end,
+                  ucp_proto_perf_factor_id_t factor_id, ucs_linear_func_t func)
+    {
+        ucs_linear_func_t funcs[1] = {func};
+        ucs_status_t status = ucp_proto_perf_add_funcs(perf.get(), start, end,
+                                                       UCS_BIT(factor_id),
+                                                       funcs, NULL);
+        ASSERT_UCS_OK(status);
```

Comment:
Is that possible to make it this way?
```suggestion
        ASSERT_UCS_OK(ucp_proto_perf_add_funcs(
                perf.get(), start, end, UCS_BIT(factor_id), &func, NULL));
```